draft-ietf-tls-tls13.md

```diff
@@ -2376,7 +2376,15 @@ RSASSA-PSS algorithms
   digest used in the mask generation function and the digest being signed are
   both the corresponding hash algorithm as defined in {{!SHS}}. When used in
   signed TLS handshake messages, the length of the salt MUST be equal to the
-  length of the digest output.  This codepoint is new in this document and is also
+  length of the digest output. When used in certificate signatures, the
+  length of the salt MUST be at least as long as the length of the
```

Comment:
the "at least" here is problematic.  I would go with equal.

Reply:
Motivation might have come from https://www.ietf.org/mail-archive/web/tls/current/msg24454.html:
> it seems that some existing implementations use weird/unusual salt
> values for RSA-PSS signature with hash algs other than SHA-1.

Don't know about such systems though, does it refer to Microsoft's implementation?

Reply:
Does it matter whether other non-TLS systems have such constraints? I'd expect any implementation to be updated anyway for TLS1.3, and conform to it.

Reply:
@Lekensteyn OpenSSL does use maximum salt length allowed for the public key size (since version 1.0.1).
Unfortunately I don't know what's the behaviour of MS implementation.

@nmav because PKI structures that use RSA-PSS signatures are already deployed and implementing any changes in them is non-trivial. If we can be compatible with them by allowing a little wiggle room (flexible salt length), I think we should do that.

Reply:
In my previous comment I missed that `When used in certificate signatures,` so my it is not  to the point.  Seeing that phrase, I think it makes sense to link to rfc5756 instead of defining formally what to expect on such certificates.